src/App.js

```diff
@@ -1,82 +1,186 @@
+                </tr>
+                <tr>
+                  <td>Mật khẩu:</td>
+                  <td>
+                    <input type="password" name="password"
+                      value={this.state.password}
+                      onChange={this.handldeChange}
+                      placeholder="Nhập Password"/>
+                  </td>
+                </tr>
+                <tr>
+                  <td>Avatar:</td>
+                  <td>
+                    <input type="file" name="file"
+                      value={this.state.file}
+                      onChange={this.handldeChange}
+                      multiple="multiple"/>
+                  </td>
+                </tr>
+                <tr>
+                  <td>Số điện thoại:</td>
+                  <td>
+                      <input type="number" name="number"
+                        value={this.state.number}
+                        onChange={this.handldeChange}
+                        placeholder="Nhập số điện thoại"/>
+                  </td>
+                </tr>
+                <tr>
+                  <td>Tuổi:</td>
+                  <td>
+                      <select name="age" value={this.state.age} onChange={this.handldeChange}>
+                          <option value="age:1">1 tuổi</option>
+                          <option value="age:2">2 tuổi</option>
+                          <option value="age:3">3 tuổi</option>
+                          <option value="age:4">4 tuổi</option>
+                          <option value="age:5">5 tuổi</option>
+                  </select>
+                  </td>
+                </tr>
+                <tr>
+                  <td>Giới tính:</td>
+                  <td>
+                      <input type="radio" name="gender" value="male"
+                        onChange={this.handldeChange}
+                        checked={this.state.gender === 'male'} />
+                          Male <br/>
+                      <input type="radio" name="gender" value="female"
+                        onChange={this.handldeChange}
+                        checked={this.state.gender === 'female'} />
+                          Female
+                  </td>
+                </tr>
+                <tr>
+                  <td>Sở thích:</td>
+                  <td>
+                      <input type="checkbox" name="hobbies"
+                        value={this.state.hobbies.duathuyen}
+                        onChange ={this.handldeChange}/>
+                          Đua thuyền <br/>
+                      <input type="checkbox" name="hobbies"
+                        value={this.state.hobbies.dabong}
+                        onChange ={this.handldeChange} />
+                          Đá bóng <br/>
+                      <input type="checkbox" name="hobbies"
+                        value={this.state.hobbies.caulong}
+                        onChange ={this.handldeChange} />
+                          Cầu lông <br/>
```

Comment:
Tạo 1 hàm để in ra các input
Bên trên đang có state là
```javascript
      this.state.hobbies = {
        sailing: {
          isChecked: false,
          text: 'Đua thuyền'
        },
        football: {
          isChecked: false,
          text: 'Đá bóng'
        },
        badminton: {
          isChecked: false,
          text: 'Cầu lông'
        }
     }
```
Thì hãy lặp `this.state.hobbies` để in ra các input
```javascript
  function generateHobbiesInput() {
    const inputs = [] // lúc đầu là 1 array rỗng, không có input nào
    const { hobbies } = this.state

    for (let key in hobbies) {
      if (hobbies.hasOwnProperty(key)) {
          /* để cho cú pháp đỡ rối, cứ phải đi gõ hobbies[key] thì gán vào biến hobby cho ngắn và cho dễ hiểu
             hơn nữa, bên dưới cũng dùng nhiều lần, nên gõ đi gõ lại hobbies[key] rất rối
          */
          const hobby = hobbies[key]

          // mỗi lần lặp các key trong hobbies thì push 1 input vào array inputs
          inputs.push(
            <div>
              <input
                type="checkbox"
                name="hobbies"
                value={hobby.isChecked}
                onChange={this.handldeChange}/>
                { hobby.text }
            </div>
          )
      }
    }
    
    return inputs
  }
```


Sau đó gọi hàm tạo các input như sau
```suggestion
                      { this.generateHobbiesInput() }
```